plugins/firebase/firebase/functions/createApp.ts

```diff
@@ -0,0 +1,43 @@
+import type * as functions from "firebase-functions/v1";
+import type {
+  Request as FunctionsRequest,
+  Response as FunctionsResponse,
+} from "firebase-functions/v1";
+import type { Hono } from "hono";
+
+interface CreateAppOptions {
+  region: string;
+  honoApp: Hono<any, any, any>;
+}
+
+export function createApp(
+  functionsInstance: typeof functions,
+  options: CreateAppOptions,
+): functions.HttpsFunction {
+  const { region, honoApp } = options;
+
+  return functionsInstance
+    .region(region)
+    .https.onRequest(async (req: FunctionsRequest, res: FunctionsResponse) => {
+      const host = req.hostname || "localhost";
+      const protocol = req.protocol || "https";
```

Comment:
It’s unlikely that requests will come from localhost, and the protocol should always be HTTPS.